extensions/pyRevitTools.extension/checks/cad_audit_check.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+
+from pyrevit import script, revit, DB, DOCS
+from pyrevit.forms import alert
+from pyrevit.preflight import PreflightTestCase
+
+from System.Windows import Window # Used for cancel button
+from rpw.ui.forms import (FlexForm, Label, Separator, Button) # RPW
+from rpw.ui.forms.flexform import RpwControlMixin, Controls # Used to create RadioButton
+
+from pyrevit.coreutils import Timer # Used for timing the check
+from datetime import timedelta # Used for timing the check
+
+doc = DOCS.doc
+ac_view = doc.ActiveView
+
+
+def collect_cadinstances(active_view_only):
+    """     Collect ImportInstance class from whole model or from just active view  """
+    if active_view_only:
+        cadinstances = DB.FilteredElementCollector(doc, ac_view.Id).OfClass(DB.ImportInstance).WhereElementIsNotElementType().ToElements()
+    else:
+        cadinstances = DB.FilteredElementCollector(doc).OfClass(DB.ImportInstance).WhereElementIsNotElementType().ToElements()
+    if cadinstances:
+        return cadinstances
+    else:
+        alert("No CAD instances found in the {}.".format("active view" if active_view_only else "model"), exitscript=True)
+
+# Manage Flexform cancel using .NET System.Windows RoutedEventArgs Class
+class ButtonClass(Window): # to handle button event (https://stackoverflow.com/questions/54756424/using-the-on-click-option-for-a-revit-rpw-flexform-button)
+    @staticmethod
+    def cancel_clicked(sender, e):
+        window = Window.GetWindow(sender)
+        window.close()
+        script.exit()
```

Comment:
I didn't tried it and I don't know rpw at all, but the problem with that SO question is that it didn't have the proper functions arguments and it called the function in the `on_click` parameters instead of just assigning them (that is, there shouldn't be parenthesis after the function name).

```suggestion
def cancel_clicked(sender, e):
    window = Window.GetWindow(sender)
    window.close()
    script.exit()
```